homeassistant/components/tolosauna/climate.py

```diff
@@ -0,0 +1,168 @@
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up climate controls for TOLO Sauna."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]["coordinator"]
+    await coordinator.async_config_entry_first_refresh()
+    async_add_entities([SaunaClimate(coordinator, entry)])
+
+
+class SaunaClimate(ToloSaunaCoordinatorEntity, ClimateEntity):
+    """Sauna climate control."""
+
+    def __init__(
+        self, coordinator: ToloSaunaUpdateCoordinator, entry: ConfigEntry
+    ) -> None:
+        """Initialize TOLO Sauna Climate entity."""
+        super().__init__(coordinator, entry)
+
+        self._attr_fan_modes = [FAN_ON, FAN_OFF]
+        self._attr_hvac_modes = [HVAC_MODE_OFF, HVAC_MODE_HEAT, HVAC_MODE_DRY]
+        self._attr_max_humidity = DEFAULT_MAX_HUMIDITY
+        self._attr_max_temp = DEFAULT_MAX_TEMP
+        self._attr_min_humidity = DEFAULT_MIN_HUMIDITY
+        self._attr_min_temp = DEFAULT_MIN_TEMP
+        self._attr_name = "Sauna Climate"
+        self._attr_precision = PRECISION_WHOLE
+        self._attr_supported_features = (
+            SUPPORT_TARGET_TEMPERATURE | SUPPORT_TARGET_HUMIDITY | SUPPORT_FAN_MODE
+        )
+        self._attr_target_temperature_step = 1
+        self._attr_temperature_unit = TEMP_CELSIUS
+        self._attr_unique_id = f"{self._config_entry.entry_id}_climate"
+
+    @property
+    def current_temperature(self) -> int:
+        """Return current temperature."""
+        return self.status.current_temperature
+
+    @property
+    def current_humidity(self) -> int:
+        """Return current humidity."""
+        return self.status.current_humidity
+
+    @property
+    def target_temperature(self) -> int:
+        """Return target temperature."""
+        return self.settings.target_temperature
+
+    @property
+    def target_humidity(self) -> int:
+        """Return target humidity."""
+        return self.settings.target_humidity
+
+    @property
+    def hvac_mode(self) -> str:
+        """Get current HVAC mode."""
+        if self.status.power_on:
+            return HVAC_MODE_HEAT
+        elif not self.status.power_on and self.status.fan_on:
+            return HVAC_MODE_DRY
+        else:
+            return HVAC_MODE_OFF
+
+    @property
+    def hvac_action(self) -> Optional[str]:
+        """Execute HVAC action."""
+        if self.status.calefaction == Calefaction.HEAT:
+            return CURRENT_HVAC_HEAT
+        elif self.status.calefaction == Calefaction.KEEP:
+            return CURRENT_HVAC_IDLE
+        elif self.status.calefaction == Calefaction.INACTIVE:
+            if self.status.fan_on:
+                return CURRENT_HVAC_DRY
+            else:
+                return CURRENT_HVAC_OFF
+        return None
+
+    @property
+    def fan_mode(self) -> str:
+        """Return current fan mode."""
+        if self.status.fan_on:
+            return FAN_ON
+        else:
+            return FAN_OFF
```

Comment:
```suggestion
        if self.status.fan_on:
            return FAN_ON
        return FAN_OFF
```

Reply:
For me it really looks better the way it is. `ON` and `OFF` have the same imporance, `OFF` is not more important than `ON`, so for me I actually find it easier to understand that there can only be two options if I explicitely `if` and `else`. Otherwise, I would have to think "are there other conditions, which may prevent from reaching `OFF`?" So I would strongly prefer to keep it this way (in this case, changed it for the other cases).

Reply:
I know what you mean, but I had this exact comment a couple of times in reviews form core developers so I thought I point it out.

Reply:
Ok... since core developers want to have it that way: Done.